pkg/cmd/render/render.go

```diff
@@ -63,6 +84,41 @@ func (r *renderOpts) Validate() error {
 		return err
 	}
 
+	clusterInfrastructureFileLen := len(r.clusterInfrastructureFile)
+	cloudProviderConfigInputFileLen := len(r.cloudProviderConfigInputFile)
+	cloudProviderConfigOutputFileLine := len(r.cloudProviderConfigOutputFile)
+
+	// Validate all files are specified when specifying infrastructure and configmap files
+	if clusterInfrastructureFileLen != 0 || cloudProviderConfigInputFileLen != 0 || cloudProviderConfigOutputFileLine != 0 {
+		if !(clusterInfrastructureFileLen != 0 && cloudProviderConfigInputFileLen != 0 && cloudProviderConfigOutputFileLine != 0) {
+			return fmt.Errorf("clulster-infrastructure-file, cloud-provider-config-input-file, and cloud-provider-config-output-file must all be specified.")
+		}
+
+		// Validate infrastructure file exists, has content, and is a regular file
+		ci, err := os.Stat(r.clusterInfrastructureFile)
+		if err != nil {
+			return err
+		}
+		if ci.Size() <= 0 {
+			return fmt.Errorf("cluster-infrastructure-file cannot be empty.")
+		}
+		if !ci.Mode().IsRegular() {
```

Comment:
would have expected opposite order: first mode, then size

Reply:
I've changed the order.